docs: add deployment.rolling doc and blog (#3530)
aws/copilot-cli: site/content/blogs/release-v118.en.md

See the section [Overview](../docs/concepts/overview.en.md) for a more detailed introduction to AWS Copilot.

## Certificate Import
_Contributed by [Penghao He](https://github.com/iamhopaul123/)_

## Ordering Deployments in a Pipeline
_Contributed by [Efe Karakus](https://github.com/efekarakus/)_

## "recreate" Strategy for Faster Redeployments
_Contributed by [Parag Bhingre](https://github.com/paragbhingre/)_

> 

Reviewer: We should mention that this applies only to LBWS, backend and WS

Author: How about faster ECS redeployment?

Author: I didn't put all the service types because when we add more ECS types in the future, it is difficult for us to remember updating here and there...

Author: Or it is probably fine because it is a blog post which is only effective within a period of time

Reviewer: Adding ECS SGTM!

!!!alert

>     Please do not set "recreate" for any services that are in production stage because of the deployment downtime.

Reviewer: nit: what do ya think of:
```suggestion
    Since "recreate" causes downtime to the service, we do **not** recommend using it for your production services.
```

b/c it's still possible (?) that users have some legit reasons to do it regardless.

> In order to support faster redeployments in developing stage, users can specify "recreate" as the deployment strategy in the service manifest:

Reviewer: What do you think of describing first "what we do today" and then "what we now offer". I actually wrote a bunch so that you can edit based on it but I lost it. So, so, so sad...

Something like "Today, a service deployment will first spin up new tasks, wait for them to be stable, and only after that will it stop the old tasks. With recreate, it will first stop the old tasks and then spin up new ones. ".

> Before v1.18, a Copilot ECS-based service (Load Balanced Web Service, Backend Service, and Worker Service) redeployment always spun up new tasks, waited for them to be stable, and then stopped the old tasks. In order to support faster redeployments for ECS-based services in developing stage, users can specify `"recreate"` as the deployment strategy in the service manifest:

Reviewer 2: maybe? there's an unfortunate repetition of "deployment," but I think it's okay
```suggestion
Before v1.18, a Copilot ECS-based service (Load Balanced Web Service, Backend Service, and Worker Service) redeployment always spun up new tasks, waited for them to be stable, and then stopped the old tasks. In order to support faster redeployments for ECS-based services in the development stage, users can specify `"recreate"` as the deployment strategy in the service manifest:
```

```yaml
deployment:
  rolling: recreate
```

> Under the hood, Copilot sets [minimumHealthyPercent and maximumPercent](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DeploymentConfiguration.html) as `0` and `100` respectively (by default we set them as `100` and `200`), so that old tasks are stopped before spinning up any new tasks.

Reviewer 3: ```suggestion
Under the hood, Copilot sets [minimumHealthyPercent and maximumPercent](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DeploymentConfiguration.html) to `0` and `100` respectively (defaults are `100` and `200`), so that old tasks are stopped before spinning up any new tasks.
```
